src/sql/workbench/services/notebook/browser/notebookServiceImpl.ts

```diff
@@ -405,7 +400,7 @@ export class NotebookService extends Disposable implements INotebookService {
 				} catch (error) {
 					this._logService.error(error);
 				}
-				instance = await this.waitOnProviderAvailability(providerDescriptor);
+				instance = await this.waitOnProviderAvailability(providerDescriptor, timeout);
```

Comment:
timeout parameter to getProviderInstance method was being left unused early which was a bug.